File: combinations.py
class Solution:
    def __init__(self):
        self.result = []

    def combine(self, n: int, k: int) -> List[List[int]]:
        self.backtrack(1,n,k,[])
        return self.result
    
    def backtrack(self,start,n,k,combination):
        if len(combination) == k:
            self.result.append(combination[:])
            return

        for i in range(start,n+1):
            self.backtrack( i+1 ,n, k , combination + [i])

File: combinations.js
/**
 * @param {number} n
 * @param {number} k
 * @return {number[][]}
 */
var combine = function(n, k) {
    
    const backtrack = (start,n,k,combination) => {
        if (combination.length == k) {
            result.push([...combination]);
            return;
        }

        for (let i=start; i<=n; i++){
            backtrack(i+1,n,k,combination.concat([i]));
        }
    }

    let result = [];
    backtrack(1,n,k,[]);
    return result;
};